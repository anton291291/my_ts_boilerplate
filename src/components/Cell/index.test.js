import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Cell } from './index';

describe('Cell', () => {
    const output = shallow(<Cell />);

    test('should render correctly', () => {
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    test('by default isClicked prop should be false', () => {
        expect(output.prop('isClicked')).toBe(false);
    });

    test('on Click the prop value should be true', () => {
        output.simulate('click');
        expect(output.prop('isClicked')).toBe(true);
    }); 
});
