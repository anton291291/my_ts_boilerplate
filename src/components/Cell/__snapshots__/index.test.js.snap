// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell should render correctly 1`] = `
<Cell__StyledCell
  isClicked={false}
  onClick={[Function]}
/>
`;
